buildpack.go

```diff
@@ -479,7 +507,13 @@ func (d *DependencyResolver) Resolve(id string, version string) (BuildpackDepend
 		return a.GreaterThan(b)
 	})
 
-	return candidates[0], nil
+	candidate := candidates[0]
+
+	if (candidate.DeprecationDate != time.Time{}) {
+		d.printDependencyDeprecation(candidate)
+	}
```

Comment:
I would like to see a test for this in `buildpack_test.go`. The section that tests the dependency resolver could do this. There are test cases now that do not set the deprecation date, so you could use one of those to validate that nothing is logged.

Then add a test case (or modify a different existing test case) to include the deprecation date & validate that output is logged.

You can pass the `bard.Logger` to the dependency resolver, your change adds that. If you pass in a `&bytes.Buffer{}` and then fetch the logged output (or lack of output) from that.